### ISSUE TITLE
Error when parsing deps

### ISSUE THREAD

**Issue by pombredanne** [xolox/python-deb-pkg-tools#9]
I encountered this error:
```
expression = 'libtdb-dev (>> 1.1) [!hurd-any]'
parse_relationship(expression)
 ValueError: Corrupt package relationship expression: Splitting name from relationship resulted in more than two tokens! (expression: libtdb-dev (>> 1.1) [!hurd-any], tokens: [u'libtdb-dev', u'>> 1.1', u'[!hurd-any]'])
```

as original found in this Ubuntu Xenial package:
```
Package: libcanberra
Binary: libcanberra0, libcanberra0-dbg, libcanberra-dev, libcanberra-pulse, libcanberra-pulse-dbg, libcanberra-gstreamer, libcanberra-gstreamer-dbg, libcanberra-gtk0, libcanberra-gtk0-dbg, libcanberra-gtk-common-dev, libcanberra-gtk-dev, libcanberra-gtk-module, libcanberra-gtk-module-dbg, libcanberra-gtk3-0, libcanberra-gtk3-0-dbg, libcanberra-gtk3-dev, libcanberra-gtk3-module, gnome-session-canberra, libcanberra-gtk3-module-dbg, libcanberra-doc
Version: 0.30-2.1ubuntu1
Priority: optional
Section: libs
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian GNOME Maintainers <pkg-gnome-maintainers@lists.alioth.debian.org>
Build-Depends: debhelper (>= 8.1.3~), cdbs (>= 0.4.93~), dh-autoreconf, gnome-pkg-tools, m4, libltdl-dev | libltdl7-dev (>= 2.2.6), dpkg-dev (>= 1.14.13), libasound2-dev [linux-any], libvorbis-dev, libglib2.0-dev (>= 2.32), libgtk-3-dev (>= 3.0.0), libgtk2.0-dev (>= 2.22), libx11-dev, libtdb-dev (>> 1.1) [!hurd-any], gtk-doc-tools, libpulse-dev (>= 0.9.11), libgstreamer1.0-dev
Architecture: any all
Standards-Version: 3.9.4
Format: 3.0 (quilt)
Directory: pool/main/libc/libcanberra
```

**Comment by pombredanne** [comment]
@xolox If I come down to this, would you feel like having a PR?

**Comment by xolox** [comment]
Hi Philippe and thanks for the feedback. There was a note about this in the documentation:

> The relationship parsing code does not understand the complete syntax defined in the Debian policy manual. More specifically architecture restrictions are not supported (because I simply don’t use them).

As implied in the note I'd never encountered a situation in which I needed to parse relationships involving architecture restrictions, because I mostly work with the control fields of binary packages (as opposed to source packages). However adding support for parsing of architecture restrictions wasn't that hard to add, so I decided to consider your feedback as a feature request and implement it now :-). I've just pushed deb-pkg-tools 4.0 to GitHub and PyPI, enjoy!
